product_variant_default_code/__manifest__.py

```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+# Copyright 2014 AvancOSC - Alfredo de la Fuente
+# Copyright 2014 Tecnativa - Pedro M. Baeza
+# Copyright 2014 Shine IT - Tony Gu
+# Copyright 2017 Tecnativa - David Vidal
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+{
+    'name': 'Product Variant Default Code',
+    'version': '10.0.1.0.0',
+    'author': 'AvancOSC,'
+              'Shine IT,'
+              'Tecnativa,'
+              'Odoo Community Association (OCA)',
+    'website': 'https://odoo-community.org/',
+    'license': 'AGPL-3',
+    'category': 'Product',
+    'depends': [
+        'product',
```

Comment:
I have an error when I install module with dependency or update module `ValueError: <type 'exceptions.TypeError'>: "Model 'sale.config.settings' does not exist in registry."`

Not sure than `sale.config.settings` are in registry with only product as dependency

Reply:
The problem here is that you are assuming that variant references are for sale. Maybe you need them for purchase only and you are forcing to install sales. Please put the configuration on the general settings wizard instead and stick to product dependency.

Reply:
Theoretically I share your vision but practically I doubt there is any company which use 'purchase' without 'sale' in the world.
And if there is one they don't use advanced 'attribute management'. But I can update my code no problem with that.
Ahahah: fucking ... config.settings transient are merged in one model in v11 : thanks for that @fpodoo